source/server/listener_manager_impl.cc

```diff
@@ -686,17 +691,30 @@ void ListenerManagerImpl::startWorkers(GuardDog& guard_dog) {
   ASSERT(!workers_started_);
   workers_started_ = true;
   uint32_t i = 0;
+
+  // We can not use "Cleanup" to simplify this logic here, because it results in a issue if Envoy is
+  // killed before workers are actually started. Specifically the AdminRequestGetStatsAndKill test
+  // case in main_common_test fails with ASAN error if we use "Cleanup" here.
+  const auto listeners_pending_init =
+      std::make_shared<std::atomic<uint64_t>>(workers_.size() * active_listeners_.size());
   for (const auto& worker : workers_) {
-    ENVOY_LOG(info, "starting worker {}", i);
+    ENVOY_LOG(info, "starting worker {}", i++);
```

Comment:
Why this change? This seems like a behavior change below for the stats?

Reply:
I have removed it for now. But I think it is not a behavioural change, but with the current code it logs "starting worker 0"  for all workers, which is confusing. I will do a separate PR for it. We can discuss there.